network/p2p/libp2pNode.go

```diff
@@ -314,33 +310,27 @@ func (n *Node) tryCreateNewStream(ctx context.Context, nodeAddress NodeAddress,
 	return s, nil
 }
 
-// GetPeerInfo generates the libp2p peer.AddrInfo for a Node/Peer given its node address
-func GetPeerInfo(p NodeAddress) (peer.AddrInfo, error) {
-	addr := MultiaddressStr(p)
+// PeerAddressInfo generates the libp2p peer.AddrInfo for an identity given its node address.
+func PeerAddressInfo(identity flow.Identity) (peer.AddrInfo, error) {
+	ip, port, key, err := networkingInfo(identity)
+	if err != nil {
+		return peer.AddrInfo{}, fmt.Errorf("could not get translate identity to networking info %s: %w", identity.NodeID.String(), err)
+	}
+
+	addr := Multiaddress(ip, port)
 	maddr, err := multiaddr.NewMultiaddr(addr)
 	if err != nil {
 		return peer.AddrInfo{}, err
 	}
-	id, err := peer.IDFromPublicKey(p.PubKey)
+
+	id, err := peer.IDFromPublicKey(key)
 	if err != nil {
-		return peer.AddrInfo{}, err
+		return peer.AddrInfo{}, fmt.Errorf("could not extract libp2p id from key:%w", err)
 	}
 	pInfo := peer.AddrInfo{ID: id, Addrs: []multiaddr.Multiaddr{maddr}}
 	return pInfo, err
 }
 
-func GetPeerInfos(addrs ...NodeAddress) ([]peer.AddrInfo, error) {
-	peerInfos := make([]peer.AddrInfo, len(addrs))
-	var err error
-	for i, addr := range addrs {
-		peerInfos[i], err = GetPeerInfo(addr)
-		if err != nil {
-			return []peer.AddrInfo{}, err
-		}
-	}
-	return peerInfos, err
-}
-
```

Comment:
This function was only getting invoked at `UpdateAllowlist`. Hence, merged the logic there.